### DISCUSSION TITLE
Exporting content from Orchard Framework and importing into Orchard Core

### DISCUSSION THREAD

**Discussion by nogginbox** [OrchardCMS/OrchardCore#7780] [Q&A] [answered]
What is the best way to move content from Orchard Framework into Orchard Core?

Orchard Framework has an Import/Export module, but I don't know if this exports into a format that can be used by Orchard Core.

Orchard Core can import content that is in a recipe. Is there a way to export a content recipe from an Orchard Framework site that can be used in an Orchard Core site?

Orchard Core can also use GraphQL. Does this provide a possible way to import content from Orchard Framework?

**Comment by Piedone** [comment, marked as answer]
Fist of all, good to see one of the elders of Orchard 1 again :).

Orchard 1 exports to XML recipes while Orchard Core consumes JSON ones. Apart from the file format, the content is quite different as well, though for basic data the mapping is quite direct (a title is still just a string, for example).

I think using import/export is the easiest way to move data between Orchard 1 and Core if you can do both operations in a fast local environment that can't time out. I'd probably create an OC-compatible JSON directly for the relevant content types from a controller in O1 or something simple as that.

Otherwise, and especially if you need to keep either of the apps online, you'd indeed need APIs. Then I'd run the export in O1 by calling APIs in OC. You don't need GraphQL for that, there's a simple RESTful API endpoint (check out `ApiController` in `OrchardCore.Contents`) that you can call to create content items.

**Reply by nogginbox** [reply]
Thanks. Think I'm either going to try an XML export and then write something to convert it to core compatible json. Or, I'll try and write a Orchard 1 controller to create the json.

I didn't think of myself as an Orchard Elder, but I have some private Orchard repositories that are almost ten years old. Doesn't seem that long ago.

**Reply by elaurentin** [reply]
@NogginBox , Did you implement a module? I was going to implement a module to (1) upload the export XML , (2) convert it back to the .json deployment recipe.
